x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_error_log.tsx

```diff
@@ -0,0 +1,275 @@
+
+  const onRefresh = () => {
+    loadEventLogs();
+  };
+
+  const columns: Array<EuiBasicTableColumn<IErrorLog>> = useMemo(
+    () => [
+      {
+        field: 'timestamp',
+        name: i18n.translate(
+          'xpack.triggersActionsUI.sections.ruleDetails.errorLogColumn.timestamp',
+          {
+            defaultMessage: 'Timestamp',
+          }
+        ),
+        render: (date: string) => (
+          <RuleEventLogListCellRenderer columnId="timestamp" value={date} dateFormat={dateFormat} />
+        ),
+        sortable: true,
+        width: '250px',
+      },
+      {
+        field: 'type',
+        name: i18n.translate('xpack.triggersActionsUI.sections.ruleDetails.errorLogColumn.type', {
+          defaultMessage: 'Type',
+        }),
+        sortable: false,
+        width: '100px',
+      },
+      {
+        field: 'message',
+        name: i18n.translate(
+          'xpack.triggersActionsUI.sections.ruleDetails.errorLogColumn.message',
+          {
+            defaultMessage: 'Message',
+          }
+        ),
+        sortable: false,
+      },
+    ],
+    [dateFormat]
+  );
+
+  useEffect(() => {
+    loadEventLogs();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dateStart, dateEnd]);
+
+  useEffect(() => {
+    const start = pagination.pageIndex * pagination.pageSize;
+    setLogList(logs.slice(start, start + pagination.pageSize));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pagination]);
+
+  useEffect(() => {
+    setLogs((prevLogs) => prevLogs.sort((a, b) => sortErrorLog(a, b, sort?.direction)));
+    setPagination((prevPagination) => ({
+      ...prevPagination,
+      pageIndex: 0,
+    }));
+  }, [sort?.direction]);
+
+  useEffect(() => {
+    if (isInitialized.current) {
+      loadEventLogs();
+    }
+    isInitialized.current = true;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [refreshToken]);
+
+  return (
+    <div>
+      <EuiSpacer />
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiSuperDatePicker
+            data-test-subj="ruleEventLogListDatePicker"
+            width="auto"
+            isLoading={isLoading}
+            start={dateStart}
+            end={dateEnd}
+            onTimeChange={onTimeChange}
+            onRefresh={onRefresh}
+            dateFormat={dateFormat}
+            commonlyUsedRanges={commonlyUsedRanges}
+            updateButtonProps={updateButtonProps}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiSpacer />
+      {isLoading && (
+        <EuiProgress size="xs" color="accent" data-test-subj="ruleEventLogListProgressBar" />
+      )}
+      <EuiBasicTable
+        data-test-subj="RuleErrorLog"
+        loading={isLoading}
+        items={logList ?? []}
+        itemId="id"
+        columns={columns}
+        sorting={{ sort }}
+        pagination={pagination}
+        onChange={({ page: changedPage, sort: changedSort }) => {
+          if (changedPage) {
```

Comment:
not really a big deal, but should we break this into `onChangePage` and `onChangeSort` functions?